eclipse-collections-code-generator/src/main/resources/api/primitiveIterable.stg

```diff
@@ -63,6 +63,13 @@ public interface <name>Iterable extends PrimitiveIterable
      */
     <type>[] toArray();
 
+    /**
+     * Converts the <name>Iterable to a primitive <type> array. If the collection fits into the provided array it is used
+     * to store its elements and is returned from the method, otherwise a new array of the appropriate size is allocated
+     * and returned. If the iterable is empty, the target array is returned unchanged.
+     */
+    <type>[] toArray(<type>[] target);
```

Comment:
If you implement this as a default method, then we can release in a minor version.